### PR TITLE
BinaryFormatter bug fix

### PR DESCRIPTION
**Pull request by adamsitnik**

### CODE REVIEW

File: src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs [hunk 1]
```diff
@@ -799,11 +799,12 @@ private long Schedule(object obj, bool assignUniqueIdToValueType, Type? type, Wr
         }
 
         // Determines if a type is a primitive type, if it is it is written
```
Comment (Tanya-Solyanik) [inline src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs:801]: `it is ` is repeated twice in the comment.

Comment (bartonjs) [reply src/libraries/System.Runtime.Serialization.Formatters/src/System/Runtime/Serialization/Formatters/Binary/BinaryObjectWriter.cs:801]: I think it's using an implicit "then".  "If it is (a primitive type, then) it is written"

**Comment by jkotas** [comment]
https://github.com/dotnet/designs/blob/main/accepted/2020/better-obsoletion/binaryformatter-obsoletion.md#all-outstanding-binaryformatter-related-issues-resolved-wont-fix-net-5 says that all binary formatter issues are going to be resolved won't fix. What makes this bug fix qualify for an exception?

**Review by JeremyKuhne — APPROVED** [review]
:shipit:

**Comment by GrabYourPitchforks** [comment]
Responding to Jan's question...

> What makes this bug fix qualify for an exception?

This was a data corruption issue which impacted a partner team and where there was no viable immediate alternative. Tactics approved a one-off exception to assist them.

**Pull request merged**